Examples/Examples/All Examples/SwiftUIExample.swift

```diff
@@ -0,0 +1,265 @@
+    /// because it's the value that contains your Mapbox access token. For that reason,
+    /// it's declared here as a `let` and is a required parameter in the initializer.
+    private let resourceOptions: ResourceOptions
+
+    init(resourceOptions: ResourceOptions, camera: Binding<Camera>) {
+        self.resourceOptions = resourceOptions
+        self._camera = camera
+    }
+
+    /// The first time SwiftUI needs to render this view, it starts by invoking `makeCoordinator()`.
+    /// SwiftUI holds on to the value you return just like it holds on to the `MapView`. This gives you a
+    /// place to direct callbacks from the MapView (delegates, observer callbacks, etc). You need to
+    /// use the coordinator for this and not this struct because this struct can be recreated many times
+    /// as your map configurations change externally. Fortunately, even as this struct is recreated,
+    /// the coordinator and the map view will only be created once.
+    func makeCoordinator() -> Coordinator {
+        Coordinator(camera: $camera)
+    }
+
+    /// After SwiftUI creates the coordinator, it creates the underlying `UIView`, in this case a `MapView`.
+    /// This method should create the `MapView`, and make sure that it is configured to be in sync
+    /// with the current settings of `SwiftUIMapView` (in this example, just the `camera` and `styleURL`).
+    func makeUIView(context: UIViewRepresentableContext<SwiftUIMapView>) -> MapView {
+        let mapView = MapView(with: .zero, resourceOptions: resourceOptions)
+        updateUIView(mapView, context: context)
+
+        /// Additionally, this is your opportunity to connect the coordinator to the map view. In this example
+        /// the coordinator is given a reference to the map view. It uses the reference to set up the necessary
+        /// observations so that it can respond to map events.
+        context.coordinator.mapView = mapView
+
+        return mapView
+    }
+
+    /// If your `SwiftUIMapView` is reconfigured externally, SwiftUI will invoke `updateUIView(_:context:)`
+    /// to give you an opportunity to re-sync the state of the underlying map view.
+    func updateUIView(_ mapView: MapView, context: Context) {
+        mapView.cameraManager.setCamera(centerCoordinate: camera.center,
+                                        zoom: camera.zoom,
+                                        animated: false)
+        /// Since changing the style causes annotations to be removed from the map
+        /// we only call the setter if the value has changed.
+        if mapView.style.styleURL != styleURL {
+            mapView.style.styleURL = styleURL
+        }
+
+        /// The coordinator needs to manager annotations because
+        /// they need to be applied *after* `.mapLoadingFinished`
+        context.coordinator.annotations = annotations
+    }
+
+    /// Here's our custom `Coordinator` implementation.
+    class Coordinator {
```

Comment:
Any reason this class is inside of a struct?

Reply:
It wouldn't have to be. Nesting it just provides some namespacing which makes it reasonable to give it a short name like `Coordinator` instead of `SwiftUIMapViewCoordinator`. It also makes a direct association between it and its containing type, which helps to communicate to developers that it's a special-purpose construct that's only designed for use with `SwiftUIMapView`.